src/Tolerance/Waiter/TimeOut.php

```diff
@@ -0,0 +1,57 @@
+<?php
+
+/*
+ * This file is part of the Tolerance package.
+ *
+ * (c) Samuel ROZE <samuel.roze@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Tolerance\Waiter;
+
+class TimeOut implements Waiter, StatefulWaiter
+{
+    /**
+     * @var Waiter
+     */
+    private $delegateWaiter;
+
+    /**
+     * @var integer
+     */
+    private $timeOut;
+
+    /**
+     * var float
+     */
+    private $secondsEllapsed;
+
+    public function __construct(Waiter $delegateWaiter, $timeOut)
+    {
+        $this->delegateWaiter = $delegateWaiter;
+        $this->timeOut = $timeOut;
+        $this->secondsEllapsed = 0;
```

Comment:
But we definitively should consider to switch the `$seconds` in `Waiter::wait` to `float` also. I currently use `0.5` in my own app. And this is why I managed `secondsEllapsed` in float